GliaWidgets/SecureConversations/SecureConversations.FileUploadList.swift

```diff
@@ -0,0 +1,965 @@
+
+        @objc private func remove() {
+            props.removeTapped()
+        }
+
+        override func layoutSubviews() {
+            super.layoutSubviews()
+            // Make FileUploadView `accessibilityFrame` smaller
+            // to allow VoiceOver to "see" `removeButton`
+            var accFrame = self.frame
+            let insetX = kContentInsets.left + kContentInsets.right + kRemoveButtonSize.width
+            accFrame.size.width -= insetX
+            accessibilityFrame = UIAccessibility.convertToScreenCoordinates(accFrame, in: self)
+        }
+
+        static func accessibleProgress(_ progress: String, to source: String?, accessibility: FileUploadStyle.Accessibility) -> String? {
+            // treat empty progress string as if it is `nil`
+            let nonEmptyProgress = progress.isEmpty ? nil : progress
+            switch (nonEmptyProgress, source) {
+            case (.none, .none):
+                return nil
+            case let (.some(percentValue), .some(fileName)):
+                return accessibility.fileNameWithProgressValue
+                    .withUploadedFileName(fileName)
+                    .withUploadPercentValue(percentValue)
+            case let (.none, .some(percentValue)):
+                return accessibility.progressPercentValue.withUploadPercentValue(percentValue)
+            case let (.some(fileName), .none):
+                return fileName
+            }
+        }
+    }
+}
+
+extension SecureConversations.FileUploadListView.FileUploadView.Props {
+    typealias FilePreviewView = SecureConversations.FileUploadListView.FileUploadView.FilePreviewView
+
+    enum State: Equatable {
+        case none
+        case uploading(Double, localFile: FilePreviewView.Kind.LocalFile)
+        case uploaded(FilePreviewView.Kind.LocalFile)
+        case error(Error)
+    }
+
+    enum Error: Equatable {
+        case fileTooBig
+        case unsupportedFileType
+        case safetyCheckFailed
+        case network
+        case generic
+
+        init(with error: CoreSdkClient.SalemoveError) {
+            switch error.error {
+            case let genericError as CoreSdkClient.GeneralError:
+                switch genericError {
+                case .networkError:
+                    self = .network
+                default:
+                    self = .generic
+                }
+            case let fileError as CoreSdkClient.FileError:
+                switch fileError {
+                case .fileTooBig:
+                    self = .fileTooBig
+                case .unsupportedFileType:
+                    self = .unsupportedFileType
+                case .infected:
+                    self = .safetyCheckFailed
+                default:
+                    self = .generic
+                }
+            default:
+                self = .generic
+            }
+        }
+    }
+}
+
+extension SecureConversations.FileUploadListView.FileUploadView {
+    class FilePreviewView: UIView {
```

Comment:
- does it make sense to move it to separate file?
- does it make sense to make it as final?
- does it make sense to inherit from BaseView?